hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java

```diff
@@ -135,17 +150,47 @@ private TransactionMetadata preHandle(
             // TODO: Verify signature via the platform once this functionality was implemented
 
             // 5. Return TransactionMetadata
-            return metadata;
+            return createTransactionMetadata(statesTracker.getUsedStates(), handlerContext);
 
         } catch (PreCheckException preCheckException) {
-            return new ErrorTransactionMetadata(
-                    txBody, preCheckException.responseCode(), preCheckException);
+            return new TransactionMetadata(txBody, payerID, preCheckException.responseCode());
         } catch (Exception ex) {
             // Some unknown and unexpected failure happened. If this was non-deterministic, I could
             // end up with an ISS. It is critical that I log whatever happened, because we should
             // have caught all legitimate failures in another catch block.
             LOG.error("An unexpected exception was thrown during pre-handle", ex);
-            return new ErrorTransactionMetadata(txBody, ResponseCodeEnum.UNKNOWN, ex);
+            return new TransactionMetadata(txBody, payerID, ResponseCodeEnum.UNKNOWN);
         }
     }
+
```

Comment:
Do we need all the below methods to be static ? Please add javadocs to the below methods

Reply:
Will do